### PR TITLE
Rayan

### PR DESCRIPTION
**Pull request by rayanfer32**
## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I ran `npm run format`/`yarn format` before commit
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings
- [ ] I have updated `develop` and merged to my branch before submitting pull request
- [ ] My pull request generate no conflicts with `develop` or `main` branch
- [ ] I requested code review from other team members

## Related Issue

- none

## Proposed Changes

- none

## Additional Info

- none

## Screenshots

|              Original              |              Updated              |
| :--------------------------------: | :-------------------------------: |
|  |  |

### CODE REVIEW

File: src/components/atoms/Rail/index.jsx [hunk 1]
```diff
@@ -1,28 +1,41 @@
...
+    <main
+      className={[props.className, styles.railContainer].join(' ')}
+      ref={railRef}>
+      {props.children}
+    </main>
+  );
+};
+
+Rail.propTypes = {
+  scrollSpeed: PropTypes.number,
```
Comment (shrivatsabhat) [inline src/components/atoms/Rail/index.jsx:36]: Scroll speed is optional
Don't give proptype as strictly required prop

File: src/components/atoms/SearchBar/index.jsx [hunk 2]
```diff
@@ -1,6 +1,17 @@
 import SearchBar from './SearchBar';
 import SearchBarLong from './SearchBarLong';
 
+import PropTypes from 'prop-types';
+
+SearchBar.propTypes = {
+  type: PropTypes.oneOf(['long']),
```
Comment (shrivatsabhat) [inline src/components/atoms/SearchBar/index.jsx:7]: Here also prop `type` and `long` are optional prop,

File: src/components/atoms/SmallCard/index.jsx [hunk 3]
```diff
@@ -2,6 +2,14 @@ import styles from './SmallCard.module.css';
 // import svgLogo from "assets/icons/icon.svg"
 import pngLogo from 'assets/icons/nexusblue.png';
 import Image from 'next/dist/client/image';
+import PropTypes from 'prop-types';
+
+SmallCard.propTypes = {
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  sublabel: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (shrivatsabhat) [inline src/components/atoms/SmallCard/index.jsx:9]: Sublabel is optional prop

**Review by shrivatsabhat — CHANGES_REQUESTED** [review]
Don't give optional prop as strictly passable proptypes

File: src/components/atoms/SearchBar/index.jsx (hunk 2 above)
Comment (rayanfer32) [reply src/components/atoms/SearchBar/index.jsx:7]: how do i do that

File: src/components/atoms/Rail/index.jsx (hunk 1 above)
Comment (rayanfer32) [reply src/components/atoms/Rail/index.jsx:36]: yes i havent made it .isRequired

**Comment by rayanfer32** [comment]
I haven't kept the props as isRequired , that means they are optional only ryt.

**Review by shrivatsabhat — APPROVED** [review]

**Pull request merged**